src/main/java/net/floodlightcontroller/meters/Meters.java

```diff
@@ -0,0 +1,216 @@
+		
+	}
+
+	@Override
+	public net.floodlightcontroller.core.IListener.Command receive(IOFSwitch sw, OFMessage msg, FloodlightContext cntx) {
+			
+		switch(msg.getType()) {
+
+			/*
+			 * We look at the FLOW_MOD instead of flow add for 2 reasons 
+			 * 	1.  We can make use of the complex matching that is already done by the controller
+			 *  2.  the FLOW_MOD sent by the controller already contains a summary of all the 
+			 *   	information from the flow that is needed for creating the meter
+			 */
+		
+			case FLOW_MOD: 	
+				// pull all the variables needed out of the flow mod 
+				OFFlowMod pi = (OFFlowMod) msg; 
+				Match testM = pi.getMatch();
+				List<OFInstruction> inst = pi.getInstructions();			
+				OFFlowModCommand type = pi.getCommand();
+				TableId table = pi.getTableId();
+				boolean corAdd = false; 
+				//logger.info("THe inputs :" + pi.toString());
+				
+				if(type == OFFlowModCommand.ADD) {
+					EthType ethtype = testM.get(MatchField.ETH_TYPE);
+					EthType ethtest = EthType.of(0x800);						
+					
+					// Need to make sure it is an action add not a meter add
+					if(inst.get(0) != null) {
+						OFInstruction testinst = inst.get(0);
+						OFInstructionType curtype = testinst.getType();	
+						OFInstructionType testval = OFInstructionType.valueOf("APPLY_ACTIONS");
+						if(curtype.equals(testval)) {
+							corAdd = true; 
+						}
+					}
+					
+					// need to check that the switch has the right factory 
+					if (sw.getOFFactory().getVersion().compareTo(OFVersion.OF_13) >= 0) {
+						factoryFlag =1;
+					}
+					
+					
+					// check that the Ethernet type add type are correct before trying to add the meter 					
+					if(ethtype == ethtest && corAdd && myflag == 1 && factoryFlag ==1) {	
+						pi.getTableId();
+						try { 
+							addMeter(testM,meterIdCounter,sw,inst, table);
+						} catch (InterruptedException e) {
+							//logger.info("We had a problem :" + pi.toString());
+							e.printStackTrace();
+						}
+						meterIdCounter ++;
+					}
+				
+				}
+			break;
+			default:
+			break;
+		}
+        return Command.CONTINUE;
+        
+    }
+	
+	
+	/*
+	 * This function created to clean up the meter add process 
+	 * takes in the information pulled from the flow mod and 
+	 * creates a meter to monitor that flow 
+	 */
+	public void addMeter(Match match, long meterID, IOFSwitch sw, List<OFInstruction> instr, TableId table) throws InterruptedException {
+		// need a list to hold all the built commands 
+		List<OFMeterBand> bands = new ArrayList<OFMeterBand>();
+		// First build the meter  
+		OFFactory meterFactory = OFFactories.getFactory(OFVersion.OF_13); 
+		OFMeterMod.Builder meterModBuilder = meterFactory.buildMeterMod().setMeterId(meterID).setCommand(OFMeterModCommand.ADD);		
+		OFMeterBand.Builder meterband = meterFactory.meterBands().buildExperimenter();
+		OFMeterBand band = meterband.build();
+        bands.add(band);
```

Comment:
this indentation is kinda triggering me

Reply:
weird it shows up as inline on my local copy